test/mockcore_test.cc

```diff
@@ -133,4 +153,71 @@ TEST_F(MockCoreFixture, RegisterAndGameStart) {
     EXPECT_EQ(gameState.mDeck.Number(), cardsInDeck);
 }
 
+TEST_F(MockCoreFixture, Draw) {
+    auto roomId = 1002;
+    std::vector<unsigned int> userIds = {11, 22, 33};
+    RegisterAndGameStart(roomId, userIds);
+
+    auto drawNum = 1;
+    EXPECT_CALL(*mMockStream, Write(
+        AllOf(NotifyMsgArgsMatcher(ErrorNumber::OK, roomId, userIds[0]),
+            DrawRspMatcher(drawNum)), _)).Times(1);
+    EXPECT_CALL(*mMockStream, Write(
+        AllOf(NotifyMsgArgsMatcher(ErrorNumber::OK, roomId, -userIds[0]),
+            DrawMatcher(drawNum)), _)).Times(1);
+    ProcessMsgFromCore(*MsgBuilder::CreateUserOperationArgs(0, roomId,
+        userIds[0], MsgBuilder::CreateDraw(drawNum)));
+
+    const auto &state = mStateMachine->GetState().mRoomIdToGameState.at(roomId);
+    auto cardsInDeck = 108 - 7 * userIds.size() - 1;
+    EXPECT_EQ(state.mDeck.Number(), cardsInDeck - 1);
+    EXPECT_EQ(state.mDiscardPile.Number(), 0);
+    EXPECT_EQ(state.mUserIdToPlayerState.at(userIds[0]).mHandcards.Number(), 8);
+    EXPECT_EQ(state.mUserIdToPlayerState.at(userIds[1]).mHandcards.Number(), 7);
+    EXPECT_EQ(state.mUserIdToPlayerState.at(userIds[2]).mHandcards.Number(), 7);
+}
+
+TEST_F(MockCoreFixture, Skip) {
+    auto roomId = 1002;
```

Comment:
how about make a private member in Fixture:  mTestRoomId = 1002.  mTestUserIds = {11, 22, 33}

Reply:
yep